SimCalorimetry/EcalSimAlgos/src/ComponentShapeCollection.cc

```diff
@@ -0,0 +1,69 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/ComponentShapeCollection.h"
+
+// #define component_shape_debug 1
+void ComponentShapeCollection::setEventSetup(const edm::EventSetup& evtSetup) {
+#ifdef component_shape_debug
+  std::cout << "ComponentShapeCollection::setEventSetup called " << std::endl;
+#endif
+  buildMe(&evtSetup);
+  for (int i=0; i<m_nDepthBins; ++i) {
+    m_shapeArr[i]->setEventSetup(evtSetup, false);
+  }
+}
+
+void ComponentShapeCollection::buildMe(const edm::EventSetup* evtSetup) {
+#ifdef component_shape_debug
+  std::cout << "ComponentShapeCollection::buildMe called " << std::endl;
+#endif
+  fillCollection(m_useDBShape);
+};
+
+void ComponentShapeCollection::fillCollection(edm::ConsumesCollector iC) {
+#ifdef component_shape_debug
+  std::cout << "ComponentShapeCollection::fillCollection(edm::ConsumesCollector iC) called " << std::endl;
+#endif
+  //m_shapeArr->clear();
+  for (int i=0; i<m_nDepthBins; ++i) {
+    m_shapeArr[i] = new ComponentShape(i, espsToken_);
```

Comment:
Where are these deleted?

Reply:
Consider using smart pointers instead.

Reply:
I changed this to `std::shared_ptr`s which I hope will automatically take care of deletes / prevent leaks.